Jenkinsfile

```diff
@@ -192,6 +192,56 @@ pipeline {
                         }
                     }
                 }
+                stage('Single node valgrind') {
+                    agent {
+                        label 'ci_vm1'
+                    }
+                    steps {
+                        provisionNodes NODELIST: env.NODELIST,
+                           node_count: 1,
+                           snapshot: true
+                        runTest stashes: [ 'CentOS-install', 'CentOS-build-vars' ],
+                                script: """set -x
+                                    . ./.build_vars-Linux.sh
+                                    CART_BASE=\${SL_PREFIX%/install*}
+                                    NODELIST=$nodelist
+                                    NODE=\${NODELIST%%,*}
+                                    trap 'set +e; set -x; ssh -i ci_key jenkins@\$NODE "set -ex; sudo umount \$CART_BASE"' EXIT
+                                    ssh -i ci_key jenkins@\$NODE "set -x
+                                        set -e
+                                        sudo mkdir -p \$CART_BASE
+                                        sudo mount -t nfs \$HOSTNAME:\$PWD \$CART_BASE
+                                        cd \$CART_BASE
+                                        ln -s /usr/bin/fusermount install/Linux/bin/fusermount3
+                                        pip3.4 install --user tabulate
+					export TR_USE_VALGRIND=memcheck
+					export IOF_TESTLOG=test/output-memcheck
+                                        nosetests-3.4 --xunit-testsuite-name=valgrind --xunit-file=nosetests-valgrind.xml --exe --with-xunit"
+                                    exit 0
+                                    """,
```

Comment:
This looks like it could be DRYed out into a common function (in the Pipeline) that can take an argument to use valgrind or not, no?

Reply:
Yes, I agree.  I was planning to merge this, then perform that as a separate step.

Reply:
I don't think I know how to do this, I could add a function like singleNodeTest() in CaRT however some of the Valgrind stuff is done in post so that wouldn't work directly.

Reply:
There is valgrind stuff done in the `post { }` in CaRT too when `singleNodeTest('')` is called with the argument `'memcheck'`.  Is the `post { }` stuff you are referring to here not the same?  Although in CaRT it's run in `post { }` and here you run it in the` steps { }`.